DarkDungeon script, FellBeast Merge Shop

Hassan0Kamel opened the pull request:
the first one will do the quests in /darkdungeon to complete the old story
the second one will farm all the items in FellBeast Merge shop, most of them are mem-only but .. yeah it's out there

Hassan0Kamel committed:
Add files via upload

Hassan0Kamel committed:
DarkDungeon script

this script will do all quests of old story darkdungeon in /darkdungeon

Hassan0Kamel committed:
FellBeast Merge Shop

this script farms all the items in FellBeast MergeShop in /darkdungeon, the other merge there is called mirror realm merge which we already have a script for

PUNK3DAF commented on Story/DarkDungeon.cs:
```diff
@@ -0,0 +1,61 @@
+            return;
+
+        Story.PreLoad(this);
+        //Bright Idea 2909
+        Story.MapItemQuest(2909, "battleoff", 17375);
+
+        //Spare Parts 2910
+        Story.MapItemQuest(2910, "battleoff", 1780, 8);
+
+        //Power It Up 2911
+        Story.KillQuest(2911, "battleoff", "Evil Moglin");
+
+        //Filthy Creatures 2912
+        Story.KillQuest(2912, "battleoff", "Evil Moglin");
```
These quests are not a part of this storyline, this is battleoff, not darkdungeon. The script for battleoff already exists.

PUNK3DAF commented on Story/DarkDungeon.cs:
```diff
@@ -0,0 +1,61 @@
+
+        //Free the Fallen 3540
+        Story.KillQuest(3540, "darkdungeon", "Dungeon Paladin");
+
```
Missing quest 3541

PUNK3DAF commented on Story/DarkDungeon.cs:
```diff
@@ -0,0 +1,61 @@
+        //The Best vs The Beast 3543
+        Story.KillQuest(3543, "darkdungeon", "Cockatrice");
+
+        //Go To Sleep 3545
+        Story.KillQuest(3545, "darkdungeon", new[] { "Dungeon Minion", "Dungeon Paladin" });
```
As I said this one is unnecessary, not part of a storyline

PUNK3DAF commented on Other/MergeShops/FellBeastMerge.cs:
```diff
@@ -0,0 +1,88 @@
+            {
+                default:
+                    bool shouldStop = Adv.matsOnly ? !dontStopMissingIng : true;
+                    Core.Logger($"The bot hasn't been taught how to get {req.Name}." + (shouldStop ? " Please report the issue." : " Skipping"), messageBox: shouldStop, stopBot: shouldStop);
```
Not sure if killing this guy is the most efficient method, might wanna change that. Good job on correctly removing the while and pickup, nicely done!

PUNK3DAF reviewed (changes requested):
Here you go

Hassan0Kamel committed:
Update DarkDungeon.cs

fixed everything and added the missing quest

Hassan0Kamel committed:
Update FellBeastMerge.cs

added the equip solo class command

Hassan0Kamel committed:
Update FellBeastMerge.cs

moved the equip class solotype to be under the logger, and capitalized the S in storyline

Hassan0Kamel committed:
Update DarkDungeon.cs

capitalized the s in storyline

PUNK3DAF committed:
Merge branch 'Skua' into Skua
